yb-voyager/src/srcdb/yugabytedb_test.go

```diff
@@ -0,0 +1,698 @@
+
+	// Expected: Only composite types (typtype='c'), NOT enums or domains
+	// hr.status_enum and hr.positive_int should be excluded
+	expectedUDTs := []string{
+		"hr.contact",
+		"inventory.device_specs",
+	}
+
+	assert.Equal(t, len(expectedUDTs), len(actualUDTs),
+		"Expected %d UDTs but got %d", len(expectedUDTs), len(actualUDTs))
+	testutils.AssertEqualStringSlices(t, expectedUDTs, actualUDTs)
+
+	// Test edge case: Empty table list
+	emptyTableList := []sqlname.NameTuple{}
+	actualEmptyUDTs := ybDB.filterUnsupportedUserDefinedDatatypes(emptyTableList)
+	assert.Equal(t, 0, len(actualEmptyUDTs), "Expected empty list for empty table list")
+}
+
+// TestYugabyteGetColumnsWithSupportedTypes_AllScenarios tests:
+// 1. UDT columns (composite types) correctly identified as unsupported with qualified names (same-schema and cross-schema)
+// 2. Extension types (hstore) in custom schemas matched against unqualified unsupported list
+// 3. Case-insensitive matching (catalog returns lowercase, unsupported list has uppercase)
+// 4. Connector type differences (GRPC vs Logical - TSVECTOR handling)
+// 5. Mixed scenarios: UDT + extension type + ENUM + regular columns in same table
+// 6. Cross-schema type usage (types from one schema used in tables in another schema)
+// 7. Offline migration mode (all types supported)
+func TestYugabyteGetColumnsWithSupportedTypes_AllScenarios(t *testing.T) {
+	testYugabyteDBSource.TestContainer.ExecuteSqls(
+		`CREATE SCHEMA custom_ext;`,
+		`CREATE SCHEMA hr;`,
+		`CREATE SCHEMA inventory;`,
+
+		`CREATE EXTENSION IF NOT EXISTS hstore WITH SCHEMA custom_ext;`,
+
+		// Create composite UDTs
+		`CREATE TYPE hr.contact AS (
+			phone VARCHAR,
+			email VARCHAR
+		);`,
+		`CREATE TYPE inventory.device_specs AS (
+			cpu VARCHAR,
+			ram INT
+		);`,
+		// Create ENUM type (supported in all modes)
+		`CREATE TYPE hr.status_enum AS ENUM ('active', 'inactive', 'pending');`,
+
+		// Case 1: Extension type in custom schema
+		// Tests that "custom_ext.hstore" from catalog matches "HSTORE" in unsupported list
+		`CREATE TABLE custom_ext.ext_table (
+			id SERIAL PRIMARY KEY,
+			name VARCHAR,
+			metadata custom_ext.hstore,
+			description TEXT
+		);`,
+
+		// Case 2: Mixed - UDT + extension type + ENUM + regular columns
+		// Tests both same-schema and cross-schema usage for UDTs: hr.contact and inventory.device_specs
+		`CREATE TABLE hr.employee_devices (
+			employee_id INT,
+			device_name VARCHAR,
+			contact_info hr.contact,
+			device_details inventory.device_specs,
+			settings custom_ext.hstore,
+			status hr.status_enum,
+			notes TEXT
+		);`,
+
+		// Case 3: TSVECTOR - only unsupported in GRPC mode
+		`CREATE TABLE custom_ext.search_table (
+			id INT PRIMARY KEY,
+			content TEXT,
+			search_vector TSVECTOR
+		);`,
+
+		// Case 4: All regular columns (baseline)
+		`CREATE TABLE hr.projects (
+			id INT PRIMARY KEY,
+			name VARCHAR,
+			budget NUMERIC
+		);`,
+	)
```

Comment:
Please add some case for case-sensitive table/column name

Reply:
I understand that it might not be a problem since we are using the catalog names from nametuples everywhere, but I still feel that one case with the case sensitivity with unsupported datatype is important for verifying that query works as expected with these.